reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClientConnect.java

```diff
@@ -119,8 +119,8 @@ public String toString() {
 
 		if (config.doOnRequestError != null) {
 			mono = mono.onErrorResume(error ->
-					Mono.subscriberContext()
-					    .doOnNext(ctx -> config.doOnRequestError.accept(new FailedHttpClientRequest(ctx, config), error))
+					Mono.deferContextual(Mono::just)
+					    .doOnNext(ctx -> config.doOnRequestError.accept(new FailedHttpClientRequest(Context.of(ctx), config), error))
```

Comment:
would it make sense for the `FailedHttpClientRequest` to accept `ContextView` instead of `Context`?

Reply:
`FailedHttpClientRequest` provides API 

```
	@Override
	public Context currentContext() {
		return context;
	}
```

So we need a `Context`